### PR TITLE
applications: serial_lte_modem: improve AT Carrier documentation

### PR DESCRIPTION
**Pull request by kacperradoszewski**
Added more descriptive documentation of custom AT commands in the LwM2M Carrier library integration.

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]

File: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst [hunk 1]
```diff
@@ -179,22 +280,38 @@ The ``<cmd>`` command is a string, and can be used as follows:
...
+* ``AT#XCARRIERCFG="carriers"[,"all"|<carrier1>[,<carrier2>[,...[,<carrier6>]]]]``
 
-  Choose the networks in which the LwM2M carrier library will apply (see the :ref:`general_options_lwm2m` section of the library's documentation).
+  Configure the networks in which the LwM2M carrier library will apply (see the :ref:`general_options_lwm2m` section of the library's documentation).
+  By default, any network is allowed. The input parameters are mapped as follows:
+
+  * ``all`` - Any network allowed.
+  * ``0`` - :kconfig:option:`CONFIG_LWM2M_CARRIER_GENERIC`.
+  * ``1`` - :kconfig:option:`CONFIG_LWM2M_CARRIER_VERIZON`.
+  * ``2`` - :kconfig:option:`CONFIG_LWM2M_CARRIER_ATT`.
```
Comment (MarkusLassila) [inline applications/serial_lte_modem/doc/CARRIER_AT_commands.rst:306]: Will conflict with https://github.com/nrfconnect/sdk-nrf/pull/14125

**Review by MarkusLassila — APPROVED** [review]

File: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst [hunk 2]
```diff
@@ -79,31 +79,127 @@ Syntax
...
+  ``<utc_time>`` must be an integer value specified in UNIX time.
+
 * ``AT#XCARRIER="velocity",<heading>,<speed_h>,<speed_v>,<uncertainty_h>,<uncertainty_v>``
+
+  Set or update the latest velocity information that will be mapped into the Velocity resource (ID:4) and Speed resource (ID:6) of the Location object (ID:6).
+  ``<heading>`` is the horizontal direction of movement in degrees clockwise from North. Must be an integer value between 0 and 359.
+  ``<speed_h>`` is the horizontal non-negative speed in meters per second. Must be a float type value in double quotes.
+  ``<speed_v>`` is the vertical speed in meters per second. Positive value indicates upward motion. Negative value indicates downward motion. Must be a float type value in double quotes.
+  ``<uncertainty_h>`` is the horizontal uncertainty speed, i.e. maximal deviation from the true speed, given in meters per second. Must be a non-negative float type value in double quotes.
+  ``<uncertainty_v>`` is the vertical uncertainty speed in meters per second. Must be a non-negative float type value in double quotes.
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CARRIER_AT_commands.rst:196]:
```suggestion
  Set or update the latest velocity information that will be mapped into the Velocity resource (ID: 4) and Speed resource (ID: 6) of the Location object (ID: 6).

  * ``<heading>`` is the horizontal direction of movement in degrees clockwise from North.
    Must be an integer value between ``0`` and ``359``.
  * ``<speed_h>`` is the horizontal non-negative speed in meters per second.
    Must be a float type value in double quotes.
  * ``<speed_v>`` is the vertical speed in meters per second.
    A positive value indicates upward motion, while a negative value indicates downward motion.
    Must be a float type value in double quotes.
  * ``<uncertainty_h>`` is the horizontal uncertainty speed in meters per second.
    Must be a non-negative float type value in double quotes.
  * ``<uncertainty_v>`` is the vertical uncertainty speed in meters per second.
    Must be a non-negative float type value in double quotes.
```

File: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst [hunk 3]
```diff
@@ -79,31 +79,127 @@ Syntax
...
+
 * ``AT#XCARRIER="voltage",<power_source>,<voltage>``
 
-The values of the parameters depend on the command string used.
-Specifying ``instance_id`` and ``resource_instance_id`` in the ``app_data`` command sends the data to the Binary App Data Container object instead of the App Data Container object.
-When using the ``app_data`` command, if no attributes are specified, SLM enters ``slm_data_mode``.
-``slm_data_mode`` is only supported when using the App Data Container object.
+  Put the value in ``<voltage>`` into the Power Source Voltage resource (ID:7) instance corresponding to one of the Available Power Sources resource (ID:6) instances of the Device object (ID:3).
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CARRIER_AT_commands.rst:200]:
```suggestion
  Put the value in ``<voltage>`` into the Power Source Voltage resource (ID: 7) instance corresponding to one of the Available Power Sources resource (ID: 6) instances of the Device object (ID: 3).
```

File: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst [hunk 4]
```diff
@@ -79,31 +79,127 @@ Syntax
...
+  Refer to the ``AT#XCARRIER="power_sources"`` command for information regarding the supported power sources.
+  ``<voltage>`` must be an integer value specified in millivolts (mV).
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CARRIER_AT_commands.rst:202]:
```suggestion
  ``<voltage>`` must be an integer value specified in millivolts.
```

File: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst [hunk 5]
```diff
@@ -179,22 +280,38 @@ The ``<cmd>`` command is a string, and can be used as follows:
...
 * ``AT#XCARRIERCFG="apn"[,<apn>]``
 
+  Configure the LwM2M carrier library to use a custom APN specified in ``<apn>`` when connecting to the device management network.
+  ``<apn>`` must be a string in double quotes.
   For details, see the :kconfig:option:`CONFIG_LWM2M_CARRIER_CUSTOM_APN` Kconfig option.
 
 * ``AT#XCARRIERCFG="auto_startup"[,<0|1>]``
 
-  Set flag to automatically apply the enabled settings to the library configuration and connect to the device management server.
+  Set flag to automatically apply the enabled settings to the LwM2M carrier library configuration and connect to the device management network.
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CARRIER_AT_commands.rst:289]:
```suggestion
  Set a flag to automatically apply the enabled settings to the LwM2M carrier library configuration and connect to the device management network.
```

File: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst [hunk 6]
```diff
@@ -179,22 +280,38 @@ The ``<cmd>`` command is a string, and can be used as follows:
...
+  * ``1`` - :kconfig:option:`CONFIG_LWM2M_CARRIER_VERIZON`.
+  * ``2`` - :kconfig:option:`CONFIG_LWM2M_CARRIER_ATT`.
+  * ``3`` - :kconfig:option:`CONFIG_LWM2M_CARRIER_LG_UPLUS`.
+  * ``4`` - :kconfig:option:`CONFIG_LWM2M_CARRIER_T_MOBILE`.
+  * ``5`` - :kconfig:option:`CONFIG_LWM2M_CARRIER_SOFTBANK`.
 
 * ``AT#XCARRIERCFG="coap_con_interval"[,<interval>]``
 
+  Configure how often the LwM2M carrier library is to send the Notify operation as a CoAP Confirmable message instead of a Non-Confirmable message.
+  ``<interval>`` must be an integer value specified in seconds. Two special values may also be used: ``-1`` to always use Confirmable notifications, or ``0`` to use the default interval of 86400 seconds.
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CARRIER_AT_commands.rst:314]:
```suggestion
  ``<interval>`` must be an integer value specified in seconds.
  Two special values may also be used: ``-1`` to always use Confirmable notifications, or ``0`` to use the default interval of 86400 seconds.
```

File: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst [hunk 7]
```diff
@@ -207,38 +324,70 @@ The ``<cmd>`` command is a string, and can be used as follows:
...
 
 * ``AT#XCARRIERCFG="hardware_version"[,<version>]``
 
+  Configure the value in ``<version>`` to be put into the Hardware Version resource (ID:18) of the Device object (ID:3).
+  ``<version>`` must be a string in double quotes.
   For details, see the :kconfig:option:`CONFIG_LWM2M_CARRIER_DEVICE_HARDWARE_VERSION` Kconfig option.
 
 * ``AT#XCARRIERCFG="manufacturer"[,<manufacturer>]``
 
+  Configure the value in ``<manufacturer>`` to be put into the Manufacturer resource (ID:0) of the Device object (ID:3).
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CARRIER_AT_commands.rst:339]:
```suggestion
  Configure the value in ``<manufacturer>`` to be put into the Manufacturer resource (ID: 0) of the Device object (ID: 3).
```

File: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst [hunk 8]
```diff
@@ -207,38 +324,70 @@ The ``<cmd>`` command is a string, and can be used as follows:
...
 * ``AT#XCARRIERCFG="service_code"[,<service_code>]``
 
+  Configure the Service Code registered for this device with LG U+.
+  ``<service_code>`` must be a string in double quotes.
   For details, see the :kconfig:option:`CONFIG_LWM2M_CARRIER_LG_UPLUS_SERVICE_CODE` Kconfig option.
 
 * ``AT#XCARRIERCFG="pdn_type"[,<pdn_type>]``
 
+  Configure the PDN type of the custom APN configured through the ``AT#XCARRIERCFG="apn"`` command.
+  ``<pdn_type>`` must be an integer value. It accepts the following values:
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CARRIER_AT_commands.rst:374]:
```suggestion
  ``<pdn_type>`` must be an integer value.
  It accepts the following values:
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Comment by kacperradoszewski** [comment]
Thanks @wiba-nordic, I believe I have applied all the suggestions.

File: applications/serial_lte_modem/doc/CARRIER_AT_commands.rst [hunk 9]
```diff
@@ -79,31 +79,151 @@ Syntax
...
+
+  Put the value in ``<data>`` into the indicated path.
+  ``<data>`` must be an opaque string in double quotes, unless ``slm_data_mode`` is enabled.
+
+  * If ``<obj_inst_id>`` and ``<res_inst_id>`` are specified, the data is set in an instance of the Data resource (ID: 0) of the Binary App Data Container object (ID: 19).
+  The URI path of the resource instance is indicated as ``/19/<obj_inst_id>/0/<res_inst_id>``.
+  * If ``<obj_inst_id>`` and ``<res_inst_id>`` are not present, the data is set in the Uplink Data resource (ID: 0) of the App Data Container object (ID: 10250).
+  The URI path of the resource instance is indicated as ``/10250/0/0``.
+  * If ``<data>`` is not present, SLM enters ``slm_data_mode`` and the data is set in the Uplink Data resource (ID: 0) of the App Data Container object (ID: 10250).
+  The URI path of the resource instance is indicated as ``/10250/0/0``.
```
Comment (wiba-nordic) [inline applications/serial_lte_modem/doc/CARRIER_AT_commands.rst:92]:
```suggestion
    The URI path of the resource instance is indicated as ``/19/<obj_inst_id>/0/<res_inst_id>``.
  * If ``<obj_inst_id>`` and ``<res_inst_id>`` are not present, the data is set in the Uplink Data resource (ID: 0) of the App Data Container object (ID: 10250).
    The URI path of the resource instance is indicated as ``/10250/0/0``.
  * If ``<data>`` is not present, SLM enters ``slm_data_mode`` and the data is set in the Uplink Data resource (ID: 0) of the App Data Container object (ID: 10250).
    The URI path of the resource instance is indicated as ``/10250/0/0``.
```

**Review by wiba-nordic — COMMENTED** [review]
A couple of spacing errors that will break the documentation build. 

I also suggest adding the following in the Updated part of the SLM section in the changelog:
```
  * The documentation for the ``#XCARRIER`` command by adding more detailed information.
```

**Review by wiba-nordic — APPROVED** [review]

**Comment by kacperradoszewski** [comment]
> A couple of spacing errors that will break the documentation build.
> 
> I also suggest adding the following in the Updated part of the SLM section in the changelog:
> 
> ```
>   * The documentation for the ``#XCARRIER`` command by adding more detailed information.
> ```

Added a slightly modified entry in the changelog to include the other command too:
`The documentation for the ``#XCARRIER`` and ``#XCARRIERCFG`` commands by adding more detailed information.`

**Pull request merged**